lib/elasticsearch.js

```diff
@@ -0,0 +1,540 @@
+
+		if(!Array.isArray(items))
+			items = [items];
+
+		for(const item of items) {
+
+			if(!item.id)
+				item.id = UUID();
+
+		}
+	}
+
+	/**
+	 * Parse the filters into elasticsearch filters
+	 * @param {Object} filters filters
+	 * @returns {Object} elasticsearch filters
+	 */
+	parseFilters(filters) {
+		return ElasticSearchFilters.getFilters(filters);
+	}
+
+	/**
+	 * Parse the item into elasticsearch script string for updates
+	 * @param {Object} item item
+	 * @returns {Object} parsed item for updates
+	 */
+	parseValuesForUpdate(item) {
+
+		let parsedItem = '';
+
+		Object.entries(item).forEach(([key, value]) => {
+			parsedItem += `ctx._source.${key}='${value}';`;
+		});
+
+		parsedItem += `ctx._source.lastModified='${new Date().toISOString()}';`;
+
+		return { script: parsedItem };
+	}
+
+	/**
+	 * Converts the items array into a multi insert bulk query for elasticsearch
+	 * @param {Array} items items
+	 * @returns {Array} parsed items
+	 */
+	parseItemsForBulkInsert(items) {
+
+		const parsedItems = [];
+
+		for(const item of items) {
+
+			parsedItems.push(
+				{ index: { _id: item.id } },
+				{ ...item, dateCreated: new Date() }
+			);
+
+		}
+
+		return parsedItems;
+	}
+
+	/**
+	 * Converts the items array into a multi upsert bulk query for elasticsearch
+	 * @param {Array} items items
+	 * @returns {Array} parsed items
+	 */
+	parseItemsForBulkUpsert(items) {
+
+		const parsedItems = [];
+
+		for(const item of items) {
+
+			parsedItems.push(
+				{ update: { _id: item.id, _type: '_doc' } },
+				{
+					doc: { ...item, lastModified: new Date() },
+					upsert: { ...item, dateCreated: new Date() }
+				}
+			);
+
+		}
+
+		return parsedItems;
+	}
+
+	/**
+	 * Parse the order params into a sort query for elasticsearch
+	 * @param {Object} order order params
+	 * @returns {Object} parsed sort query
+	 */
+	parseSortingParams(order) {
+
+		const parsedSorting = Object.entries(order).map(([key, value]) => {
+
+			return {
+				[`${key}.raw`]: value
+			};
+
+		});
+
+		return { sort: parsedSorting };
+	}
+
+	/**
+	 * Inserts data into elasticsearch
+	 * @param {Model} model Model instance
+	 * @param {Object} item item
+	 * @returns {Boolean} true/false
+	 */
+	async insert(model, item) {
+
+		this.validateModel(model);
+		this.prepareFields(item);
+
+		item.dateCreated = new Date();
+
+		try {
+			const client = this.getClient();
+			const res = await client.create({
+				index: model.constructor.table,
+				type: '_doc',
+				id: item.id,
+				refresh: 'wait_for',
+				body: item
+			});
+
+			return res.body.result === 'created';
+
+		} catch(err) {
+			throw new ElasticSearchError(err.message, ElasticSearchError.codes.ELASTICSEARCH_ERROR);
+		}
+	}
+
+	/**
+	 * Inserts multiple items into elasticsearch
+	 * @param {Model} model Model instance
+	 * @param {Array} items items
+	 * @returns {Boolean} true/false
+	 */
+	async multiInsert(model, items) {
+
+		this.validateModel(model);
+		this.prepareFields(items);
```

Comment:
Se podria hacer que retorne el valor modificado en vez de modificarlo por dentro
```suggestion
		item = this.prepareFields(item);
```